tl/features/context_match.py

```diff
@@ -0,0 +1,332 @@
+import pandas as pd
+import re
+import sys
+import os
+
+class match(object):
+    def __init__(self, input_path, context_path, args):
+        #print(args)
+        self.final_data = pd.read_csv(input_path)
+        self.data = pd.DataFrame()
+        self.result_data = pd.DataFrame()
+        self.context = pd.read_csv(context_path)
+        self.output_column_name = args.get("output_column")
+        self.threshold_1 = args.pop("sim_string")
+        self.threshold_2 = args.pop("sim_quantity")
+        self.debug = args.pop("debug")
+        self.final_data = self.final_data.reindex(columns = self.final_data.columns.tolist() 
+                                  + ['context_properties', 'monge_elkan_sim', self.output_column_name])
+    def preprocess(self, sent):
+        sent = sent.lower()
+        new_sent = self.remove_punctuation(sent)
+        new_sent = self.tokenize(new_sent)
+        return new_sent
+
+    def tokenize(self, sent):
+        sent = sent.split(" ")
+        return sent
+
+    def remove_punctuation(self, sent):
+        res = re.sub(r'[^\w\s]', '', sent)
+        return res
+
+    def quantity_matching(self, context_target, context_try):
+        if context_target == 0.0 and context_try == 0.0:
+            return 1
+        max_val = max(abs(context_target), abs(context_try))
+        abs_diff = abs(context_target - context_try)
+        final_val = 1 - (abs_diff/max_val)
+        return final_val
+
+    def remove_comma(self, string):
+        string = string.replace(',', '')
+        return string
+
+    def matching_with_quantity(self, context_to_do, all_props, check = "q"):
+        res = [idx for idx in all_props if idx.lower().startswith(check.lower())]
+        prop_val = "NaN"
+        sim = 0.0
+        res2 = "NaN"
+        for x in res:
+            x = x.split(":")
+            n = x[0]
+            check_with_t = n[1:]

```

Comment:
This whole loop is messed up. As I said before please use reasonable parameter names. Line 53, 54, 55 and prime examples of bad parameters name